interfaces/README.md

```diff
@@ -143,3 +143,95 @@ Run these commands from every app's own 'root'-folder.
+
+- Testing global variables and objects, when are being defined or constructed
+- There are several methods which serve the purpose of defining class wide variables, which we should also test and verify. One of the typical examples of one such method is `ngOnInit`
+        
+        it('ngOnInit: should set up variables', () => {
+          expect(component.isLoggedIn).toBeDefined(); // check for class variables to be defined
+          expect(component.someValye).toBeTruthy();  // check for a variable to be TRUE
+          expect(component.someValye).toBeFalsy();  // check for a variable to be FALSE
+        });
+
+#### Testing method callbacks and changes
+-  By utilizing the `spy` provided within the jasmine framework, we should always test and verify that the appropriate methods have been called.
+
+        it('some_method: should call another fn', () => {
+          spyOn(event, "preventDefault"); // Monitor the said method
+          component.doLogin(event); // call some_method
+          expect(event.preventDefault).toHaveBeenCalled(); // check for the monitored method to have been called
+        });
+
+#### Testing conditional statements
+-  Make separate `it` blocks for different conditions.
+
+        it("Test when xyz === 'somethinf'") ....
+        it("Test when xyz !== 'somethinf'") ....
+
+#### Testing conditional statements
+-  Make separate `it` blocks for different conditions.
+
+        it("Test when xyz === 'somethinf'") ....
+        it("Test when xyz !== 'somethinf'") ....
+
+
+#### Testing Async Mehthods (i.e. methods which make an API call)
+-  Make a Spy for the specific async call which returns a Promise object. For example a method containing a call routine `this.programsService.changePassword` can be spied using following
+
+        let spy = spyOn(component.programsService, "changePassword").and.returnValue(Promise.resolve(true));
+
+
+-  Based on the changes / executions upon the completion of the async request, we should aim to test the changes and modifications.
+
+        spy.calls.mostRecent().returnValue.then(() => { // block to test what happenes after the async calls.
+            // Here goes expectations and changes
+        });
+
+-  Make sure the `done()` method is used to account for the async calls and fake async stubs/spies.
+
+        it('XYZ', (done) => {
+          // spies and stubs
+
+          spy.calls.mostRecent().returnValue.then(() => {
+              // tests
+              done(); // to complete the tests
+          });
+        });
+
+#### Testing HTML Tags
+-  By using the `defaultEl` and the monitoring the changes within the HTML pages. However, the testing here does not bring a lot of productivity in terms of what we get out of it. So, we can choose to discard this aspect of testing.
+-  HTML tags are tested by matching the `string` values, which is not very intuitive with `i18n` modules in use
```

Comment:
In our current set-up of the tests, the translations are **not** actually loaded, so the strings that are present in the HTML during the test-runs are only the 'keys' for the actual translation. So the string-value would be `'page.personal.pageTitle'`, instead of `'Personal page'`, for example.